tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard.bal

```diff
@@ -1748,6 +1748,43 @@ function yoo(SomeRecord|int|() aa) {
 
 }
 
+type R1 readonly & record {|int i;|}?;
+type R2 (readonly & record {|int i;|})?;
+type R record {|int i;|};
+
+function testR1(R1 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR2(R2 r) returns boolean {
+    return r is record {|int i;|};
+}
```

Comment:
This needs to be in the type test tests. The ones below with the`is` expression in an if statement are the ones that should be in the type guard tests here.

Reply:
Changed